### ISSUE TITLE
В связи с приостановкой активной разработки

### ISSUE THREAD

**Issue by Cookies1488** [mavrikkk/ha_kettler#69]
Уточните, планируется ли поддержка текущего функционала (в случае обновления home assistant какие-то вещи устареют)?

**Comment by mavrikkk** [comment]
если это можно будет исправить вслепую (что пока с переменным успехом удается вот уже год), то да.
А если говорить совсем глобально, то проект всегда был, есть и будет открытым. Никому не мешаю форкать и распространять и/или делать пулл реквесты улучшений, исправлений.

**Comment by Drafteed** [comment]
@mavrikkk а на что перешли если не секрет?)

**Comment by mavrikkk** [comment]
никакого секрета. Если кратко, то ни на что ))

Если подробно, то...
У меня был все время только 1 чайник G200S, который попал ко мне можно сказать случайно. Параллельно настраивал HA, так и возникла идея интеграции. Пока чайник служил мне верой и правдой я всячески дорабатывал и развивал модуль. По иронии судьбы так получилось, что модуль стал широко известным только после смерти моего чайника. Многие устройства серии Redmond я добавлял в модуль вслепую. В итоге понял, что это сложно и неправильно, отнимает много времени, и не всегда конечный результат приносит удовольствие ) После этого появилось данное сообщение.

Вообще идей для доработок полно:
1. Нашел несколько вариантов асинхронных блютус библиотек, надо бы проверить, может какая нативно в HassOS заведется без танцев с бубном.
2. После этого можно и нужно переписать модуль на выбранную библиотеку.
3. Практически нерешенным остался вопрос с интеграцией нескольких устройств одновременно. В этом направлении есть несколько идей, пробовать которые надо на практике с отладкой процесса.
4. У кого то было предложение по несколько иной группировке объектов внутри интеграции. Из этой идеи у меня родилась неплохая возможная доработка.
5. Другая компиляция вызова функций. Тестировал здесь с кем-то. Очень удачная и быстрая оказалась. В релиз так и не вошла.

Без устройства на руках это все не проверить. Возможно, найдутся энтузиасты, которые доделают/переделают данный модуль.

Сам пока не нашел что-либо нужное мне из продукции компании Redmond. В последнее время больше увлекся zigbee протоколом. Также открыл свой репозиторий HassIO addons. Посмотреть можно в профиле. Пока там из полезного upnp openhome dlna media player (локальный медиа проигрыватель, если на устройстве есть аудио выход). Позволяет проигрывать музыку с любого dlna сервера (в тч со смартфона) прямо на HassOS с поддержкой плейлистов, блекджека ну и всего остального, кроме airplay (мне не интересно).

**Comment by NigaMan** [comment]
проект реально офигенный, только начал изучать и уже подключил чайник и все работает, но при подключении чайника к серверу постоянно моргает светодиод на чайнике, это как-то можно пофиксить?

**Comment by Drafteed** [comment]
> 1. Нашел несколько вариантов асинхронных блютус библиотек, надо бы проверить, может какая нативно в HassOS заведется без танцев с бубном.

Если честно я бы превратил это в mqtt конвертер, по типу zigbee2mqtt или `ESP32-R4sGate-for-Redmond` в образе докера. А внутри уже можно использовать что душе угодно :)

**Comment by mavrikkk** [comment]
> проект реально офигенный, только начал изучать и уже подключил чайник и все работает, но при подключении чайника к серверу постоянно моргает светодиод на чайнике, это как-то можно пофиксить?

Да. Это называется режим активного сопряжения...уже сам не помню точно )) но точно помню, что можно выбрать эту функцию при первоначальной настройке.
Вам нужно удалить чайник, Добавить его заново. При добавлении сменить true на false в нужном пункте.

**Comment by mavrikkk** [comment]
> > 1. Нашел несколько вариантов асинхронных блютус библиотек, надо бы проверить, может какая нативно в HassOS заведется без танцев с бубном.
> 
> Если честно я бы превратил это в mqtt конвертер, по типу `zigbee2mqtt` или `ESP32-R4sGate-for-Redmond` в образе докера. А внутри уже можно использовать что душе угодно :)

Идея отличная, однако она не отменяет блютус составляющей ) и ее асинхронности ))
Однако, если забыть про компоненту и сделать доккер аддон, то да, многие ограничения снимаются. Нужно лишь правильно приготовить образ и настроить все службы внутри, но привязка к библиотекам HassOS действительно пропадет. Задача интересная и классная! 
Возьметесь? )

**Comment by Drafteed** [comment]
> Возьметесь? )

Я бы с радостью, но у меня всего один чайник :)
